Assets/Controllers/CameraMotionHandler.cs

```diff
@@ -44,7 +48,7 @@ public void MoveToPosition(Vector3 position)
     {
         position.y = this.transform.position.y;
         position.z += -5;
-        this.transform.position = position;
+        newPosition = position;
```

Comment:
You may want to pass this in as something like `currentPosition`, position and newPosition is a bit confusing to read. Also, the staticly set -5 might be better off as a constant or config based setting somewhere.